libexec/kvm-cluster-up/user-data

```diff
@@ -4,6 +4,14 @@ growpart:
   mode: auto
   devices: ['/']
 
+users:
```

Comment:
I don't understand why this is necessary. I get a cloud username for fedora, rhel, centos, debian already.

Reply:
If I want to set the username to something of my choice, it doesn't work correctly.

Reply:
There's already code in libexec/kvm-cluster-up/user-data that tries to set the password for KUP_CLOUD_USERNAME; that code takes an error if KUP_CLOUD_USERNAME is set to anything but the default for the distribution.

Reply:
By reusing KUP_CLOUD_USERNAME (and the key here is **CLOUD**) I think we're conflating its value. Are you after additional user accounts?

Reply:
I wanted to add a specified user account instead of the default, in my use case.

Reply:
Given my experience with cloud-init you want to do the absolute minimum and let something else add additional roles (e.g., ansible). The reason I "hid" all this stuff in libexec was to prevent feature creep. I could see having additional packages being installed as part of the initial boot but I backed that out last week[1] because the package names may a) not be available and b) likely to be named ever-so-slightly differently across the distros. The goal here was to facilitate booting a machine as simply and as quickly is possible; no surprises!

I think you would be better served by using ansible to build on top of whatever your cloud image yields.

[1] https://github.com/frobware/fed-k8s/commit/84d4f8647581883e87813e217e167d05ce3df7da